source/_posts/2018-01-16-Testing-Vuejs-in-Rails-with-Webpacker-and-Jest.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: post
+title: "Testing Vue.js in Rails with Webpacker and Jest"
+category:
+comments: true
+author:
+  name: Mario Saul
+  email: mario@wyeworks.com
+  twitter_handle: mario_saul
+  github_handle:  mariiio
+  image:  /images/team/mario-saul.jpg
+  description: Software Developer at Wyeworks.
+---
+
+In the project I'm working on, I was given the task of investigating how to integrate [Vue.js](https://vuejs.org/) with our existing Rails app. So I started reading the official guide, watching tutorials and reading various posts until I finally got a full working Vue component.
+
+Finally I was ready to write some tests. But unfortunately, the [Webpacker](https://github.com/rails/webpacker) gem doesn't include testing, so I had to do the configuration on my own.
+
+To my surprise, I found that there wasn't much documentation on how to do the setup. So I figured I’d make this post to share with you how I managed to eventually get it working.
+
+<!--more-->
+
+## 1. Install Jest
+I decided to go for Jest for no personal preference, I just noticed that [vue-cli](https://github.com/vuejs/vue-cli) ships with it and went for it.
+
+To install [Jest](https://facebook.github.io/jest/) all you have to do is run `yarn add --dev jest` from the root of your project.
+Then, add a test script to your `package.json`:
+{% codeblock lang:json %}
+{
+  "scripts": {
+    "test": "jest",
+    ...
+  },
+  ...
+}
+{% endcodeblock %}
+Now you can run your tests with `yarn test`.
+
+## 2. Define the location of your tests
+If you try to run `yarn test` at this point, you'll see that `config/webpack/test.js` failed. This is because the way [Jest looks for test files in the project](https://facebook.github.io/jest/docs/en/configuration.html#testmatch-array-string). It basically executes all files matching `.spec.js` or `.test.js` in the whole project. In this case, the file matched `*.test.js` so it tried to run it as a test.
+
+Since we don't want the Webpack config file as well as other files in the project that meet this criteria to run with our tests, we need to tell Jest where to look for them.
+
+In my case since I'm using [Rspec](http://rspec.info/), I decided to point it to the `spec/javascripts` directory. But you can feel free to choose whichever directory fits your project the best.
```

Comment:
choose whichever directory fits your project the best. => choose whichever directory better fits your project